### PR TITLE
chore: uppercase the liquibase table names

### PR DESCRIPTION
**Pull request by jkwatson-verta**
This is to account for a bug in liquibase when working with an H2 database.

### CODE REVIEW

**Review by eheinlein-verta — COMMENTED** [review]
There is code in UAC that expects the table name to be lowercased (https://github.com/VertaAI/verta-uacservice/blob/013d8424b06144d7968b766a9957b4b5c847e612/src/main/java/ai/verta/uac/App.java#L644) specifically because we switched from the old name to the new one that was being used in modeldb. Does this take that into account? (By which I mean has it been tested in UAC, and if need be is there a separate PR for UAC?)

**Comment by jkwatson-verta** [comment]
> There is code in UAC that expects the table name to be lowercased (https://github.com/VertaAI/verta-uacservice/blob/013d8424b06144d7968b766a9957b4b5c847e612/src/main/java/ai/verta/uac/App.java#L644) specifically because we switched from the old name to the new one that was being used in modeldb. Does this take that into account? (By which I mean has it been tested in UAC, and if need be is there a separate PR for UAC?)

Tables names are not supposed to be case-sensitive (you can always query against either case in every RDBMS I've ever come across), which is why I believe there is a bug in liquibase about this. If we're dependent on table name casing, then we also have a bug.

It looks like UAC already deals with this, though: https://github.com/VertaAI/verta-uacservice/blob/main/src/main/java/ai/verta/uac/App.java#L599-L601

**Review by eheinlein-verta — APPROVED** [review]

**Comment by jkwatson-verta** [comment]
!rebuild

**Comment by jkwatson-verta** [comment]
!rebuild

**Comment by jkwatson-verta** [comment]
> > There is code in UAC that expects the table name to be lowercased (https://github.com/VertaAI/verta-uacservice/blob/013d8424b06144d7968b766a9957b4b5c847e612/src/main/java/ai/verta/uac/App.java#L644) specifically because we switched from the old name to the new one that was being used in modeldb. Does this take that into account? (By which I mean has it been tested in UAC, and if need be is there a separate PR for UAC?)
> 
> Tables names are not supposed to be case-sensitive (you can always query against either case in every RDBMS I've ever come across), which is why I believe there is a bug in liquibase about this. If we're dependent on table name casing, then we also have a bug.
> 
> It looks like UAC already deals with this, though: https://github.com/VertaAI/verta-uacservice/blob/main/src/main/java/ai/verta/uac/App.java#L599-L601

I decided to verify this worked before I merged, and good thing I did. It looks like Amazon Aurora actually has case-sensitive table names, so this totally broke things. I updated it to only uppercase when H2 is the database. Thanks for spotting that, @eheinlein-verta , as that would have been a really bad thing to merge!

File: backend/common/src/main/java/ai/verta/modeldb/common/CommonDBUtil.java [hunk 1]
```diff
@@ -288,9 +287,16 @@ protected void runLiquibaseMigration(
...
-    System.getProperties().put("liquibase.databaseChangeLogTableName", "database_change_log");
-    System.getProperties()
-        .put("liquibase.databaseChangeLogLockTableName", "database_change_log_lock");
+    String changeLogTableName = "database_change_log";
+    String changeLogLockTableName = "database_change_log_lock";
+    if (config.getRdbConfiguration().isH2()) {
+      // H2 upper cases all table names, and liquibase has issues if you don't make this also upper
+      // case.
+      changeLogLockTableName = changeLogLockTableName.toUpperCase();
+      changeLogTableName = changeLogTableName.toUpperCase();
```
Comment (jkwatson-verta) [inline backend/common/src/main/java/ai/verta/modeldb/common/CommonDBUtil.java]:
```suggestion
      changeLogTableName = changeLogTableName.toUpperCase();
      changeLogLockTableName = changeLogLockTableName.toUpperCase();
```

**Review by eheinlein-verta — APPROVED** [review]

**Pull request merged**